### PR TITLE
bind to localhost. Fixes #1400

### LINKED ISSUE #1400: Insecure Default Bind to 0.0.0.0
After analyzing the idea behind why `0.0.0.0` is being used, I still didn't understand why a person that didn't read the docs closely will have to expose their MobSF to the world because it's an insecure default.

Yes, if they're running a container, that could be a case. That's the only case.

If someone is running it on a server, the application should still be secure and make the admin update it.
I know I can update it in the configuration file, I am just asking for the `default insecure` option on a security tool.

### PR DESCRIPTION
**Pull request by superpoussin22**
### Describe the Pull Request

Refer to issue #1400 

### Checklist for PR

- [X] Run MobSF unit tests and lint `tox -e lint,test`.
- [X] Tested Working on Linux, Mac,  and Docker
- [X] Make sure build is passing on your PR. [Build Status](https://travis-ci.com/MobSF/Mobile-Security-Framework-MobSF/pull_requests)

### Additional Comments (if any)

```
Not tested on windows 
Doc must be updated to say it run on localhost only by default
```

### CODE REVIEW

**Comment by ajinabraham** [comment]
@superpoussin22 This might go un-noticed on an update. There are folks who use run script in LAN/Public servers. I was thinking the following approach and document it accordingly in docs.
`run.sh` - will bind to 0.0.0.0
`run.sh 127.0.0.1:8000`

What say?

**Comment by superpoussin22** [comment]
@ajinabraham agree with you it would be better

File: run.bat [hunk 1]
```diff
@@ -1,2 +1,2 @@
 @echo off
-.\venv\Scripts\activate && waitress-serve --listen=*:8000 --threads=10 --channel-timeout=1800 MobSF.wsgi:application
+.\venv\Scripts\activate && waitress-serve --listen=127.0.0.1:8000 --threads=10 --channel-timeout=1800 MobSF.wsgi:application
```
Comment (ajinabraham) [inline run.bat:2]: Keep `--listen` be default as *:8000. Allow user to set `ip:port` as first argument for localhost and document it as default in docs.

File: run.sh [hunk 2]
```diff
@@ -2,4 +2,4 @@
 # Dev Server
 #. venv/bin/activate && python manage.py runserver
 # Prod Server
-. venv/bin/activate && gunicorn -b 0.0.0.0:8000 MobSF.wsgi:application --workers=1 --threads=10 --timeout=1800
\ No newline at end of file
+. venv/bin/activate && gunicorn -b 127.0.0.1:8000 MobSF.wsgi:application --workers=1 --threads=10 --timeout=1800
```
Comment (ajinabraham) [inline run.sh:5]: Keep `-b` default as 0.0.0.0:8000. Allow user to set `ip:port` as first argument for localhost and document it as default in docs.

**Comment by superpoussin22** [comment]
do you want a test also on  port ?

File: run.sh [hunk 3]
```diff
@@ -2,4 +2,50 @@
...
 # Prod Server
-. venv/bin/activate && gunicorn -b 0.0.0.0:8000 MobSF.wsgi:application --workers=1 --threads=10 --timeout=1800
\ No newline at end of file
+var="$1"
+
+function validate_ip () {
+local IP=$1
+local stat=1
+
+if [[ $IP =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
```
Comment (ajinabraham) [inline run.sh:11]: This is nice!! Can windows batchfile support any validations like this?

Comment (superpoussin22) [reply run.sh:11]: need to check for windows :)

**Comment by superpoussin22** [comment]
we could adapt : 
https://stackoverflow.com/questions/20299266/ip-verification-in-batch-script-first-match-by-findstr-secondly-verify-by-for

**Comment by ajinabraham** [comment]
@superpoussin22 You got a chance to look into this?

File: run.sh [hunk 4]
```diff
@@ -1,5 +1,48 @@
...
+    OIFS=$IFS
```
Comment (ajinabraham) [inline run.sh:9]: @superpoussin22 What is this assignment doing?

**Pull request merged**